src/placeholder/inputs/parameters.py

```diff
@@ -0,0 +1,142 @@
+from dataclasses import fields, field, InitVar
+from pydantic.dataclasses import dataclass
+from typing import List, Callable
+import numpy as np
+
+@dataclass
+class Variable:
+
+    covariate: str
+    var_link_fun: Callable
+    fe_init: float
+    re_init: float
+    re_zero_sum_std: float = field(default=np.inf)
+    fe_gprior: List[float] = field(default_factory=lambda: [0.0, np.inf])
+    re_gprior: List[float] = field(default_factory=lambda: [0.0, np.inf])
+    fe_bounds: List[float] = field(default_factory=lambda: [-np.inf, np.inf])
+    re_bounds: List[float] = field(default_factory=lambda: [-np.inf, np.inf])
+
+    def __post_init__(self):
+        assert isinstance(self.covariate, str)
+        assert len(self.fe_gprior) == 2
+        assert len(self.re_gprior) == 2
+        assert len(self.fe_bounds) == 2
+        assert len(self.re_bounds) == 2
+        assert self.fe_gprior[1] > 0.0
+        assert self.re_gprior[1] > 0.0
+
+
+@dataclass
+class Parameter:
+
+    param_name: str
+    link_fun: Callable
+    variables: InitVar[List[Variable]]
+
+    num_fe: int = field(init=False)
+    covariate: List[str] = field(init=False)
+    var_link_fun: List[Callable] = field(init=False)
+    fe_init: List[float] = field(init=False)
+    re_init: List[float] = field(init=False)
+    re_zero_sum_std: List[float] = field(init=False)
+    fe_gprior: List[List[float]] = field(init=False)
+    re_gprior: List[List[float]] = field(init=False)
+    fe_bounds: List[List[float]] = field(init=False)
+    re_bounds: List[List[float]] = field(init=False)
+
+    def __post_init__(self, variables):
+        assert isinstance(variables, list)
+        assert len(variables) > 0
+        assert isinstance(variables[0], Variable)
+        self.num_fe = len(variables)
```

Comment:
This won't be true if the `variables` has `SplineVariable` in it.

Reply:
@zhengp0 we'll use your utility function here when you have it :)

Reply:
Here it is for now :) b846138

Reply:
Thanks!